core/storage/blog/gae_models.py

```diff
@@ -334,16 +334,25 @@ def has_reference_to_user_id(cls, user_id):
             cls.editor_ids == user_id).get(keys_only=True) is not None
 
     @classmethod
-    def get_by_user(cls, user_id):
+    def get_by_user(cls, user_id, published=False, limit=None):
         """Retrieves the rights object for all blog posts assigned to given user
 
         Args:
             user_id: str. ID of user.
+            published: bool. True if rights model for published blog post are to
+                be fetched else false.
+            limit: int. Number of BlogPostRightsModel to be fetched. All
+                existing models will be fetched if none.
 
         Returns:
             list(BlogPostRightsModel). The list of BlogPostRightsModel objects
             in which the given user is a editor.
         """
+        if limit:
+            return cls.query(
+                cls.editor_ids == user_id,
+                cls.blog_post_is_published == published
+                ).order(-cls.last_updated).fetch(limit)
         return cls.query(cls.editor_ids == user_id).fetch()
```

Comment:
Why is "published" not applied when there is no limit? 

There should be tests for all 4 cases (2 × 2) of the function arguments.

Reply:
I did not see any use case where we will be required to fetch all published blog posts rights models or all draft blog post rights models, currently. We will be fetching all blog post rights models both published and drafts for user (eg when removing  blog post editor role for author) or will be specifying the limit. So, did not add published when there is no limit.

Reply:
When you're writing a method at the storage layer, you can't just tailor it to the functionality you happen to be implementing at the time. This is because you're writing a general method that may be used by others in the future who would reasonably expect it to work for all args that they throw at it. This is the art of API design and if you design this only around your particular use case (or, worse, design it to "look" more general but have it only satisfy your particular use cases) then it makes the codebase more fragile and this is something that should be avoided.

So, either make this function fully accurate in all cases (even the cases which you don't need), or have multiple functions that are individually fully accurate.

Reply:
Added 3 methods replacing this one -- get_multi_published_models_by_user(user_id, limit) , get_multi_draft_models(user_id, limit), get_all_by_user(user_id).